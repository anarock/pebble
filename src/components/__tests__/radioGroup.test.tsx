import * as React from "react";
import { Radio, RadioGroup } from "../";
import { mount } from "enzyme";
import sinon from "sinon";
import combos from "combos";
import renderer from "react-test-renderer";
import { cx } from "emotion";
import {
  setWrap,
  selectedButton,
  unSelectedButton
} from "./fixtures/radioSet.style";

function noop() {}

describe("RadioGroup", () => {
  test("should call onChange on click with correct arguments", () => {
    const spy = sinon.spy();

    const radio = mount(
      <RadioGroup selected={"radio-2"} onChange={spy} name="test">
        <Radio value="radio-1" label="I am radio button" />
        <Radio value="radio-2" label="I am radio button" />
      </RadioGroup>
    );

    radio
      .find(Radio)
      .at(1)
      .simulate("click");
    expect(spy.called).toBeFalsy();

    radio
      .find(Radio)
      .at(0)
      .simulate("click");

    expect(spy.calledWith("radio-1")).toBeTruthy();
  });

  test("should call onChange on click with correct arguments", () => {
    const spy = sinon.spy();

    const radio = mount(
      <RadioGroup selected={"radio-1"} onChange={spy} name="test" toggle>
        <Radio value="radio-1" label="I am radio button" />
        <Radio value="radio-2" label="I am radio button" />
      </RadioGroup>
    );

    radio
      .find(Radio)
      .at(0)
      .simulate("click");

    expect(spy.calledWith(undefined)).toBeTruthy();
  });
});

describe("set of radios", () => {
  const _props = {
    value: ["ON", "OFF"]
  };
  const _combos = combos(_props);

  test.each(_combos)("%o", props => {
    const toggle = renderer.create(
      <RadioGroup
        selected={props.value}
        onChange={noop}
        name="set of radios"
        className={setWrap}
      >
        <Radio
          value="OFF"
          label=""
          className={cx({
            [unSelectedButton]: true,
            [selectedButton]: props.value === "OFF"
          })}
        >
          {() => "OFF"}
        </Radio>
        <Radio
          value="ON"
          label=""
          className={cx({
            [unSelectedButton]: true,
            [selectedButton]: props.value === "ON"
          })}
        >
          {() => "ON"}
        </Radio>
      </RadioGroup>
    );
    const tree = toggle.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
