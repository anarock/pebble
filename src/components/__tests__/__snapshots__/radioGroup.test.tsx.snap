// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`set of radios { value: 'OFF' } 1`] = `
.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  padding: 10px 0;
  position: relative;
  font-weight: 400;
  color: #101721;
  font-size: 14px;
}

.emotion-1[data-disabled='true'] {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-1[data-disabled='true'] .css-qatnln {
  color: #E0E0E0;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 400;
  color: #101721;
  font-size: 12px;
}

.emotion-2 {
  border: 1px solid #EDEDED;
  border-radius: 0px 3px 3px 0px;
  padding: 10px 20px;
}

.emotion-0 {
  border: 1px solid #EDEDED;
  border-radius: 3px 0px 0px 3px;
  padding: 10px 20px;
  background-color: #EDEDED;
}

<div
  aria-label="set of radios"
  className="emotion-4"
  role="radiogroup"
>
  <div
    aria-checked={true}
    className="emotion-1"
    onClick={[Function]}
    role="radio"
    tabIndex={0}
  >
    <div
      className="emotion-0"
    >
      OFF
    </div>
  </div>
  <div
    aria-checked={false}
    className="emotion-1"
    onClick={[Function]}
    role="radio"
    tabIndex={-1}
  >
    <div
      className="emotion-2"
    >
      ON
    </div>
  </div>
</div>
`;

exports[`set of radios { value: 'ON' } 1`] = `
.emotion-1 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  padding: 10px 0;
  position: relative;
  font-weight: 400;
  color: #101721;
  font-size: 14px;
}

.emotion-1[data-disabled='true'] {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-1[data-disabled='true'] .css-qatnln {
  color: #E0E0E0;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: 400;
  color: #101721;
  font-size: 12px;
}

.emotion-0 {
  border: 1px solid #EDEDED;
  border-radius: 3px 0px 0px 3px;
  padding: 10px 20px;
}

.emotion-2 {
  border: 1px solid #EDEDED;
  border-radius: 0px 3px 3px 0px;
  padding: 10px 20px;
  background-color: #EDEDED;
}

<div
  aria-label="set of radios"
  className="emotion-4"
  role="radiogroup"
>
  <div
    aria-checked={false}
    className="emotion-1"
    onClick={[Function]}
    role="radio"
    tabIndex={-1}
  >
    <div
      className="emotion-0"
    >
      OFF
    </div>
  </div>
  <div
    aria-checked={true}
    className="emotion-1"
    onClick={[Function]}
    role="radio"
    tabIndex={0}
  >
    <div
      className="emotion-2"
    >
      ON
    </div>
  </div>
</div>
`;
